src/store/auth/reducer.tsx

```diff
@@ -28,106 +28,96 @@ const INIT_STATE = {
         })
       );
-    })
-  )
-  .handleAction(
-    authActions.nukeUsersSuccess,
-    produce((draft) => {
+    });
+  })
+  .handleAction(authActions.nukeUsersSuccess, (state) => {
+    return produce(state, (draft) => {
       draft.users = mapResult(draft.users, () => []);
-    })
-  )
+    });
+  })
   .handleAction(
     authActions.setAllowDuplicateEmailsSuccess,
-    produce((draft, { payload }) => {
-      draft.allowDuplicateEmails = payload;
-    })
+    (state, { payload }) => {
+      return produce(state, (draft) => {
+        draft.allowDuplicateEmails = payload;
+      });
+    }
   )
-  .handleAction(
-    authActions.setUserDisabledSuccess,
-    produce((draft, { payload }) => {
+  .handleAction(authActions.setUserDisabledSuccess, (state, { payload }) => {
+    return produce(state, (draft) => {
       draft.users = mapResult(draft.users, (users: AuthUser[]) => {
         return users.map((u: AuthUser) =>
           u.localId === payload.localId
             ? { ...u, disabled: payload.disabled }
             : u
         );
       });
-    })
-  )
-  .handleAction(
-    authActions.deleteUserSuccess,
-    produce((draft, { payload }) => {
+    });
+  })
+  .handleAction(authActions.deleteUserSuccess, (state, { payload }) => {
+    return produce(state, (draft) => {
       draft.users = mapResult(draft.users, (users: AuthUser[]) =>
         users.filter((u: AuthUser) => u.localId !== payload.localId)
       );
-    })
-  )
-  .handleAction(
-    authActions.updateFilter,
-    produce((draft, { payload: { filter } }) => {
-      draft.filter = filter;
-    })
-  )
-  .handleAction(
-    authActions.openAuthUserDialog,
-    produce((draft, { payload: authUserDialogData }) => {
-      draft.authUserDialogData = authUserDialogData;
-    })
-  )
-  .handleAction(
-    authActions.setAuthUserDialogLoading,
-    produce((draft, { payload: loading }) => {
+    });
+  })
+  .handleAction(authActions.updateFilter, (state, { payload }) => {
+    return produce(state, (draft) => {
+      draft.filter = payload.filter;
+    });
+  })
+  .handleAction(authActions.openAuthUserDialog, (state, { payload }) => {
+    return produce(state, (draft) => {
+      draft.authUserDialogData = payload;
+    });
+  })
+  .handleAction(authActions.setAuthUserDialogLoading, (state, { payload }) => {
+    return produce(state, (draft) => {
       if (draft.authUserDialogData) {
-        draft.authUserDialogData.loading = loading;
+        draft.authUserDialogData.loading = payload;
       }
-    })
-  )
-  .handleAction(
-    authActions.setAuthUserDialogError,
-    produce((draft, { payload: error }) => {
+    });
+  })
+  .handleAction(authActions.setAuthUserDialogError, (state, { payload }) => {
+    return produce(state, (draft) => {
       if (draft.authUserDialogData && draft.authUserDialogData.result) {
-        draft.authUserDialogData.result.error = error;
+        (draft.authUserDialogData.result as any).error = payload;
```

Comment:
why do you need any here?
Is this because of the readonly error?
https://immerjs.github.io/immer/typescript#cast-utilities